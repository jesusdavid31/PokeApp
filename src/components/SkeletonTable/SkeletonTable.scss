/* SkeletonTable.scss */
$sk-bg: #f2f3f5;
$sk-base: #e6e8eb;
$sk-highlight: #f5f7fa;
$sk-border: #e3e5e8;

.sk-shimmer {
    position: relative;
    overflow: hidden;
    background: linear-gradient(
        90deg,
        $sk-base 0%,
        $sk-highlight 50%,
        $sk-base 100%
    );
    background-size: 200% 100%;
    animation: sk-shine 1.3s linear infinite;
    border-radius: 6px;
}

@keyframes sk-shine {
    0% { background-position: 200% 0; }
    100% { background-position: -200% 0; }
}

.sk-table {
    background: #fff;
    border: 1px solid $sk-border;
    border-radius: 12px;
    padding: 16px;
    display: grid;
    gap: 12px;
    margin-top: 15px;

    &__header {
        display: flex;
        align-items: center;
        justify-content: space-between;

        .sk-table__title {
            width: 160px;
            height: 20px;
        }

        .sk-table__actions {
            display: flex;
            gap: 8px;
        }
    }

    &__frame {
        border: 1px solid $sk-border;
        border-radius: 10px;
        overflow: hidden;

        table {
            width: 100%;
            border-collapse: collapse;

            thead th {
                background: $sk-bg;
                padding: 12px;
                border-bottom: 1px solid $sk-border;
            }

            tbody td {
                padding: 12px;
                border-bottom: 1px solid $sk-border;
            }

            tfoot td {
                padding: 12px;
                background: #fff;
            }
        }
    }
}

.sk-cell {
    display: inline-block;
    height: 12px;
    width: 100%;
    max-width: 140px;

    // Variar longitudes para naturalidad
    &:nth-child(1) { max-width: 110px; }
    &:nth-child(2) { max-width: 90px; }
    &:nth-child(3) { max-width: 120px; }
    &:nth-child(4) { max-width: 80px; }
    &:nth-child(5) { max-width: 100px; }
    &:nth-child(6) { max-width: 70px; }
    &:nth-child(7) { max-width: 130px; }
}

.sk-btn {
    height: 32px;
    width: 96px;
    border-radius: 8px;

    &.sm {
        height: 32px;
        width: 72px;
    }

    &.lg {
        width: 200px;
        height: 32px;
    }
}

.sk-pagination {
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 12px;

    .sk-page-dots {
        display: flex;
        gap: 8px;

        .sk-dot {
            width: 28px;
            height: 32px;
            border-radius: 8px; // rectangulitos tipo botón de página
        }
    }
}

/* Responsivo básico */
@media (max-width: 640px) {
    .sk-table__header .sk-table__title { width: 120px; }
    .sk-btn { width: 80px; }
    .sk-cell { max-width: 100px; }
}
