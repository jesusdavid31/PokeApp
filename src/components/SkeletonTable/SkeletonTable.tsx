
// Estilos
import "./SkeletonTable.scss";

export default function SkeletonTable() {
    const cols = 7;
    const rows = 10;

    return (
        <div className="sk-table">

            <div className="sk-table__frame">
                <table>
                    <thead>
                        <tr>
                            {Array.from({ length: cols }).map((_, i) => (
                                <th key={i}>
                                    <span className="sk-cell sk-shimmer" />
                                </th>
                            ))}
                        </tr>
                    </thead>

                    <tbody>
                        {Array.from({ length: rows }).map((_, r) => (
                            <tr key={r}>
                                {Array.from({ length: cols }).map((_, c) => (
                                    <td key={c}>
                                        <span className="sk-cell sk-shimmer" />
                                    </td>
                                ))}
                            </tr>
                        ))}
                    </tbody>

                    <tfoot>
                        <tr>
                            <td colSpan={cols}>
                                <div className="sk-pagination">
                                    <div className="sk-page-dots">
                                        <div className="sk-btn lg sk-shimmer" />
                                    </div>
                                    <div className="sk-page-dots">
                                        <div data-testid="button" className="sk-btn sm sk-shimmer" /> {/* Prev */}
                                        <span className="sk-dot sk-shimmer" />
                                        <span className="sk-dot sk-shimmer" />
                                        <span className="sk-dot sk-shimmer" />
                                        <span className="sk-dot sk-shimmer" />
                                        <div className="sk-btn sm sk-shimmer" /> {/* Next */}
                                    </div>
                                </div>
                            </td>
                        </tr>
                    </tfoot>
                </table>
            </div>
        </div>
    );
}
