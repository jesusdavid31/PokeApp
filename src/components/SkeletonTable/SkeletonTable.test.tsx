import { render, screen } from '@testing-library/react';
import SkeletonTable from './SkeletonTable';

test('renders SkeletonTable correctly', () => {
    const { container } = render(<SkeletonTable />);  // Obtenemos el container directamente de render()

    // Verifica si la tabla está presente
    const tableElement = screen.getByRole('table');
    expect(tableElement).toBeInTheDocument();

    // Verifica si la cabecera de la tabla (thead) tiene 7 celdas
    const headerCells = screen.getAllByRole('columnheader'); // "th" son los headers
    expect(headerCells.length).toBe(7); // Esperamos que haya 7 celdas en el encabezado

    // Verifica si el cuerpo de la tabla (tbody) tiene 10 filas, y cada fila tiene 7 celdas de "sk-shimmer"
    const rows = screen.getAllByRole('row'); // Todos los tr, incluidas las filas de los datos
    expect(rows.length).toBe(12); // 1 fila de encabezado + 10 filas de datos


    // Verifica si la paginación está presente en el pie de la tabla (tfoot)
    const pagination = container.querySelector('.sk-pagination'); // Usamos querySelector en el container
    expect(pagination).toBeInTheDocument(); // Verifica que el div con la clase "sk-pagination" esté en el DOM

    // Verifica si los botones de paginación están presentes
    const paginationButtons = screen.getAllByTestId('button');
    expect(paginationButtons.length).toBeGreaterThan(0); // Al menos un botón de paginación
});
