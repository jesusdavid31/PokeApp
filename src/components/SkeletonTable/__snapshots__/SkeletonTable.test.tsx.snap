// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`snapshot 1`] = `
<div>
  <div
    class="sk-table"
  >
    <div
      class="sk-table__frame"
    >
      <table>
        <thead>
          <tr>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
            <th>
              <span
                class="sk-cell sk-shimmer"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
          <tr>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
            <td>
              <span
                class="sk-cell sk-shimmer"
              />
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colspan="7"
            >
              <div
                class="sk-pagination"
              >
                <div
                  class="sk-page-dots"
                >
                  <div
                    class="sk-btn lg sk-shimmer"
                  />
                </div>
                <div
                  class="sk-page-dots"
                >
                  <div
                    class="sk-btn sm sk-shimmer"
                  />
                   
                  <span
                    class="sk-dot sk-shimmer"
                  />
                  <span
                    class="sk-dot sk-shimmer"
                  />
                  <span
                    class="sk-dot sk-shimmer"
                  />
                  <span
                    class="sk-dot sk-shimmer"
                  />
                  <div
                    class="sk-btn sm sk-shimmer"
                  />
                   
                </div>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
